File: week2/pset/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    // Prompt the user for some text
    string text = get_string("Text: ");

    // Count the number of letters, words, and sentences in the text
    int letters = 0;
    int words = 1;
    int sentences = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (('A' <= text[i] && text[i] <= 'Z') || ('a' <= text[i] && text[i] <= 'z'))
        {
            letters++;
        }
        if (text[i] == ' ')
        {
            words++;
        }
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }

    // Compute the Coleman-Liau index
    float l = (float) letters / words * 100;
    float s = (float) sentences / words * 100;

    float index = 0.0588 * l - 0.296 * s - 15.8;

    // Print the grade level
    if (index >= 1 && index < 16)
    {
        printf("Grade %i\n", (int) round(index));
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Before Grade 1\n");
    }
}

File: week6/pset/sentimental-readability/readability.py
from cs50 import get_string


def main():
    # Prompt the user for some text
    text = get_string("Text: ")

    # Count the number of letters, words, and sentences in the text
    letters = 0
    words = 1
    sentences = 0
    for i in range(len(text)):
        if 'A' <= text[i] and text[i] <= 'Z' or 'a' <= text[i] and text[i] <= 'z':
            letters += 1
        elif text[i] == ' ':
            words += 1
        elif text[i] == '.' or text[i] == '!' or text[i] == '?':
            sentences += 1

    l = letters / words * 100
    s = sentences / words * 100;

    index = 0.0588 * l - 0.296 * s - 15.8

    if index >= 1 and index < 16:
        print(f"Grade {round(index)}")
    elif index >= 16:
        print("Grade 16+")
    else:
        print("Before Grade 1")


main()
